src/services/plugin.service.ts

```diff
@@ -82,7 +107,7 @@ export class PluginService {
     });
   }
 
-  public setPluginState(plugin: string, active: boolean) {
+  public setPluginState(plugin: string, active: boolean): IPlugin {
```

Comment:
I don't like that this function fetches and sets the state of a plugin. You should prob make another function to fetch a plugin, then a `void` function to `setPluginState`

Reply:
The reason that this function is wrapped like this is because, whenever a state change occurs in plugin, it also needs to be saved. This isolates plugins from having to manually save plugin states. Basically, the business logic is abstracted out.

Reply:
My bad, that's actually not the case, I'll go ahead move the save logic into there though.